.github/workflows/test.yml

```diff
@@ -220,6 +228,86 @@ jobs:
+      - uses: actions/checkout@v4
+      - name: Run in netbsd
+        uses: vmactions/netbsd-vm@v1
+        with:
+          usesh: true
+          prepare: |
+            /usr/sbin/pkg_add -v bash
+            ftp https://go.dev/dl/go${{matrix.go}}.netbsd-amd64.tar.gz
+            mkdir /usr/local
+            rm -fr /usr/local/go && tar -C /usr/local -xf go${{matrix.go}}.netbsd-amd64.tar.gz
+            ln -s /usr/local/go/bin/go /usr/local/bin
+          run: |
+            echo "Running tests on $(uname -a) at $PWD"
+            
+            export PATH=$PATH:/usr/local/go/bin/
+
+            # verify Go is available
+            go version
+
+            echo "=> go build"
+            go build -v ./...
+            # Compile without optimization to check potential stack overflow.
+            # The option '-gcflags=all=-N -l' is often used at Visual Studio Code.
+            # See also https://go.googlesource.com/vscode-go/+/HEAD/docs/debugging.md#launch and the issue hajimehoshi/ebiten#2120.
+            go build "-gcflags=all=-N -l" -v ./...
+
+            # Check cross-compiling Windows binaries.
+            env GOOS=windows GOARCH=386 go build -v ./...
+            env GOOS=windows GOARCH=amd64 go build -v ./...
+            env GOOS=windows GOARCH=arm go build -v ./...
+            env GOOS=windows GOARCH=arm64 go build -v ./...
+
+            # Check cross-compiling macOS binaries.
+            env GOOS=darwin GOARCH=amd64 go build -v ./...
+            env GOOS=darwin GOARCH=arm64 go build -v ./...
+
+            # Check cross-compiling Linux binaries.
+            env GOOS=linux GOARCH=amd64 go build -v ./...
+            env GOOS=linux GOARCH=arm64 go build -v ./...
+
+            # Check cross-compiling FreeBSD binaries.
+            env GOOS=freebsd GOARCH=amd64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...
+            env GOOS=freebsd GOARCH=arm64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...
+            
+            # Check cross-compiling NetBSD binaries.
+            env GOOS=netbsd GOARCH=amd64 go build -v ./...
+            env GOOS=netbsd GOARCH=arm64 go build -v ./...
+
+            echo "=> go mod vendor"
+            mkdir /tmp/vendoring
+            cd /tmp/vendoring
+            go mod init foo
+            echo 'package main' > main.go
+            echo 'import (' >> main.go
+            echo '  _ "github.com/ebitengine/purego"' >> main.go
+            echo ')' >> main.go
+            echo 'func main() {}' >> main.go
+            go mod edit -replace github.com/ebitengine/purego=$GITHUB_WORKSPACE
+            go mod tidy
+            go mod vendor
+            go build -v .
+
+            cd $GITHUB_WORKSPACE
+            echo "=> go test CGO_ENABLED=0"
+            env CGO_ENABLED=0 go test -shuffle=on -v -count=10 ./...
+
+            echo "=> go test CGO_ENABLED=1"
+            env CGO_ENABLED=1 go test -shuffle=on -v -count=10 ./...
```

Comment:
I was wondering if it would be possible to integrate the script for FreeBSD and NetBSD.

Reply:
So do you do this in the next PR?

Reply:
I'm currently looking into seeing if they can be merged into one